precompile/contracts/deployerallowlist/config.go

```diff
@@ -0,0 +1,58 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package deployerallowlist
+
+import (
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/precompile/allowlist"
+	"github.com/ava-labs/subnet-evm/precompile/config"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var _ config.Config = &Config{}
+
+// Config wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type Config struct {
+	allowlist.Config
+	config.Uprade
+}
+
+// NewConfig returns a config for a network upgrade at [blockTimestamp] that enables
+// ContractDeployerAllowList with [admins] and [enableds] as members of the allowlist.
+func NewConfig(blockTimestamp *big.Int, admins []common.Address, enableds []common.Address) *Config {
+	return &Config{
+		Config: allowlist.Config{
+			AdminAddresses:   admins,
+			EnabledAddresses: enableds,
+		},
+		Uprade: config.Uprade{BlockTimestamp: blockTimestamp},
+	}
+}
+
+// NewDisableConfig returns config for a network upgrade at [blockTimestamp]
+// that disables ContractDeployerAllowList.
+func NewDisableConfig(blockTimestamp *big.Int) *Config {
+	return &Config{
+		Uprade: config.Uprade{
+			BlockTimestamp: blockTimestamp,
+			Disable:        true,
+		},
+	}
+}
+
+func (Config) Key() string { return ConfigKey }
+
+// Equal returns true if [cfg] is a [*ContractDeployerAllowListConfig] and it has been configured identical to [c].
+func (c *Config) Equal(cfg config.Config) bool {
+	// typecast before comparison
+	other, ok := (cfg).(*Config)
+	if !ok {
+		return false
+	}
+	return c.Uprade.Equal(&other.Uprade) && c.Config.Equal(&other.Config)
+}
+
+func (c *Config) Verify() error { return c.Config.Verify() }
```

Comment:
Should we remove this since `c.Config` is embedded?

Reply:
Marking as resolved since it seems reasonable to keep this explicit as most other precompile configs have further verification to do.